libbroker/broker/internal/peering.hh

```diff
@@ -63,6 +63,11 @@ public:
     return removed_;
   }
 
+  /// Returns the removal reason, which may be empty.
+  std::string removed_reason() const noexcept {
+    return removed_reason_;
+  }
```

Comment:
This cannot be `noexcept` because it can thow bad_alloc. However, there's no reason to copy the string in the first place:

```suggestion
  const std::string& removed_reason() const noexcept {
    return removed_reason_;
  }
```